tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,376 @@
+            addl_hints: PartialRecordsHints = {}
+            pandas_compatible_addl_hints: PartialRecordsHints = {
+                'datetimeformat': f'{dateformat} HH24:MI:SS',
+                'datetimeformattz': f'{dateformat} HH:MI:SS',
+            }
+            uses_pandas = False
+            if self.engine.name == 'redshift':
+                if dateformat != 'YYYY-MM-DD':
+                    # this is the only format supported by Redshift on
+                    # export
+                    uses_pandas = True
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                addl_hints = pandas_compatible_addl_hints
+            elif self.engine.name == 'postgresql':
+                if dateformat == 'YYYY-MM-DD':
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'datetimeformattz': 'YYYY-MM-DD HH:MI:SS',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                    }
+                else:
+                    uses_pandas = True
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if dateformat != 'YYYY-MM-DD':
+                    uses_pandas = True
+            elif self.engine.name == 'bigquery':
+                # All current export is via Avro
+                uses_pandas = True
+            if uses_pandas:
+                # we're going to need to be sure to use hints that work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'dateformat': dateformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload(column_name='date',
+                                       records_format=records_format)
+                self.assertEqual(csv_text, create_sample(dateformat) + "\n",
+                                 f"from dateformat {dateformat} and addl_hints {addl_hints}")
+            except ModuleNotFoundError as e:
+                if 'pandas' in str(e) and expect_pandas_failure:
+                    # as expected
+                    continue
+                else:
+                    raise
+            self.assertFalse(expect_pandas_failure)
+
+    def test_unload_datetime(self) -> None:
```

Comment:
C901 'RecordsUnloadDatetimeIntegrationTest.test_unload_datetime' is too complex (17)